src/app/public/modules/dock/dock.service.ts

```diff
@@ -0,0 +1,208 @@
+
+    if (insertBeforeElement) {
+      this.renderer.insertBefore(bottomDockElement, subjectElement, insertBeforeElement);
+    } else {
+      this.renderer.appendChild(bottomDockElement, subjectElement);
+    }
+  }
+
+  /**
+   * Removes an element reference from the bottom dock.
+   * @param elementRef
+   */
+  public removeFromBottom(elementRef: ElementRef): void {
+    /* istanbul ignore else */
+    if (this.bottomDockRef) {
+      this.renderer.removeChild(this.bottomDockElement, elementRef.nativeElement);
+      if (this.bottomDockElement.children.length === 0) {
+        this.destroyBottomDock();
+      }
+    }
+  }
+
+  private createBottomDock(): void {
+    this.bottomDockRef = this.dynamicComponentService.createComponent(
+      SkyDockComponent
+    );
+  }
+
+  private destroyBottomDock(): void {
+    this.bottomDockObserver.disconnect();
+    this.ngUnsubscribe.next();
+
+    if (this.bottomDockStyleElement) {
+      this.destroyBottomStyleElement();
+    }
+
+    this.renderer.removeChild(document.body, this.bottomDockElement);
+
+    this.bottomDockHeight =
+    this.bottomDockObserver =
+    this.bottomDockRef =
+    this.bottomDockStyleElement =
+    this.ngUnsubscribe = undefined;
+  }
+
+  private destroyBottomStyleElement(): void {
+    this.renderer.removeChild(document.head, this.bottomDockStyleElement);
+  }
+
+  private adjustBottomMargin(): void {
+    const dockHeight = this.bottomDockElement.getBoundingClientRect().height;
+    if (dockHeight === this.bottomDockHeight) {
+      return;
+    }
+
+    // Create a style element to avoid overwriting any existing inline body styles.
+    const styleElement = this.renderer.createElement('style');
+    const textNode = this.renderer.createText(`body { margin-bottom: ${dockHeight}px; }`);
+
+    // Apply a `data-` attribute to make unit testing easier.
+    this.renderer.setAttribute(
+      styleElement,
+      'data-test-selector',
+      'sky-layout-dock-bottom-styles'
+    );
+
+    this.renderer.appendChild(styleElement, textNode);
+    this.renderer.appendChild(document.head, styleElement);
+
+    if (this.bottomDockStyleElement) {
+      this.destroyBottomStyleElement();
+    }
+
+    this.bottomDockHeight = dockHeight;
+    this.bottomDockStyleElement = styleElement;
+  }
+
+  private findInsertSibling(stackOrder: number): Element {
+    const childElements = this.bottomDockElement.children;
+    for (let i = 0, len = childElements.length; i < len; i++) {
+      const childElement = childElements.item(i);
+      const currentStackOrder = +childElement.getAttribute(ATTR_STACK_ORDER);
+      if (!isNaN(currentStackOrder) && stackOrder >= currentStackOrder) {
+        return childElement;
+      }
+    }
+  }
+
+  private getHighestStackOrder(): number {
+    const childElements = this.bottomDockElement.children;
+
+    let lastStackOrder = -Infinity;
+    for (let i = 0, len = childElements.length; i < len; i++) {
```

Comment:
If there are no child elements, the highest stack order will always return `-Infinity`. Do you want a final check for that before you return?